### PR TITLE
Sketch for implementation of Context

### PR DESCRIPTION
**Pull request by Adrijaned**
For now just a showcase for possible implementation

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by Cervator** [comment]
Looks sensible to me, although I'm not getting into the deep details just yet. Pinging @vampcat and @eviltak for review and feedback on applying Context and related config stuff to a new project. Wondering if FlexibleConfig would be relevant here, or if DS might make a good implementation challenge beyond the plans in Terasology

File: engine/src/main/java/org/destinationsol/game/SolGame.java [hunk 1]
```diff
@@ -113,15 +114,15 @@
     private HullConfig respawnHull;
     private boolean isPlayerRespawned;
 
-    public SolGame(SolApplication cmp, String shipName, boolean tut, boolean isNewGame, CommonDrawer commonDrawer) {
-        solApplication = cmp;
+    public SolGame(SolApplication cmp, String shipName, boolean tut, boolean isNewGame, CommonDrawer commonDrawer, Context context) {
+        this.context = context;
```
Comment (eviltak) [inline engine/src/main/java/org/destinationsol/game/SolGame.java]: A context should not be cached in a field; instead, a context must be passed to the constructor and the instances required from the context must be retrieved and cached in fields. In this case, the `SolApplication` instance (used later in the class) in the context must be retrieved (`context.get(SolApplication.class)`) and cached in a field.

File: engine/src/main/java/org/destinationsol/game/screens/MainScreen.java [hunk 2]
```diff
@@ -90,7 +93,8 @@
     private final TextPlace myChargesExcessTp;
     private final TextPlace myMoneyExcessTp;
 
-    public MainScreen(float resolutionRatio, RightPaneLayout rightPaneLayout, SolApplication solApplication) {
+    public MainScreen(float resolutionRatio, RightPaneLayout rightPaneLayout, Context context, SolApplication solApplication) {
+        this.context = context;
```
Comment (eviltak) [inline engine/src/main/java/org/destinationsol/game/screens/MainScreen.java]: Same as above.

File: engine/src/main/java/org/destinationsol/game/sound/OggSoundManager.java [hunk 3]
```diff
@@ -39,10 +41,12 @@
     private final Map<String, OggSound> soundMap;
     private final Map<SolObject, Map<OggSound, Float>> loopedSoundMap;
     private final DebugHintDrawer debugHintDrawer;
+    private final Context context;
 
     private float myLoopAwait;
 
-    public OggSoundManager() {
+    public OggSoundManager(Context context) {
+        this.context = context;
```
Comment (eviltak) [inline engine/src/main/java/org/destinationsol/game/sound/OggSoundManager.java]: Same as above.

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Pull request merged**